### ISSUE TITLE
bad headers matching rules v2

### ISSUE THREAD

**Issue by vigohe** [DiUS/pact-jvm#786]
I have a problem when matching headers

```
"$.headerContent-Type": {
    "match": "regex",
    "regex": "application/json;\\s?charset=(utf|UTF)-8"
}
```

should be `$.headers.Content-Type`

**Comment by uglyog** [comment]
Can you provide some more context for this issue? Pact-JVM version and the test code that resulted in this JSON fragment?

**Comment by vigohe** [comment]
artifact: `pact-jvm-provider-maven_2.12` version: `3.5.11` 

Using `PactDslJsonBody`

 `.headers(responseHeaders).matchHeader("Content-Type", "application/json;\\s?charset=(utf|UTF)-8")`

does generate 

```
"$.headerContent-Type": {
    "match": "regex",
    "regex": "application/json;\\s?charset=(utf|UTF)-8"
}
```
And the header should be `$.headers.Content-Type` not `$.headerContent-Type`

**Comment by uglyog** [comment]
Released 3.6.0-rc.1 with this fix.
